EIPS/eip-3540.md

```diff
@@ -79,58 +81,95 @@ The container starts with the EOF header:
-5. Section data size MUST be equal to `section_size` declared in its header.
-6. Stray bytes outside of sections MUST NOT be present. This includes trailing bytes after the last section.
+3. There MUST be at least one section (and therefore section header).
+4. Section content size MUST be equal to size declared in its header.
+5. Stray bytes outside of sections MUST NOT be present. This includes trailing bytes after the last section.
 
 ### EOF version 1
 
-#### Section kinds
+EOF version 1 is made up of 5 EIPs, including this one: [EIP-3540](./eip-3540.md), [EIP-3670](./eip-3670.md), [EIP-4200](./eip-4200.md), [EIP-4750](./eip-4750.md), and [EIP-5450](./eip-5450.md). Some values in this specification are only discussed briefly. To understand the full scope of EOF, it is necessary to review each EIP in-depth.
+
+#### Container
+
+The EOF version 1 container consists of a `header` and `body`.
+
+```
+magic      := hex"ef00"
+version    := 0x01
+terminator := 0x00
+
+container  := header, body
+header := magic, version, kind_type, type_size, kind_code, num_code_sections, code_size+, kind_data, data_size, terminator
+body := type_section, code_section+, data_section
+type_section := (inputs, outputs, max_stack_height)+
+```
+
+*note: `,` is a concatenation operator and `+` should be interpreted as "one or more" of the preceding item*
+
+##### Header
 
-The section kinds for EOF version 1 are defined as follows. The list may be extended in future versions.
+| name              | length   | value         | description                                                                        |
+|-------------------|----------|---------------|------------------------------------------------------------------------------------|
+| magic             | 2 bytes  | 0xEF00        | EOF prefix                                                                         |
+| version           | 1 byte   | 0x01          | EOF version                                                                        |
+| kind_type         | 1 byte   | 0x01          | kind marker for EIP-4750 type section header                                       |
+| type_size         | 2 bytes  | 0x0003-0xFFFF | uint16 denoting the length of the type section content                             |
+| kind_code         | 1 byte   | 0x02          | kind marker for code size section                                                  |
+| num_code_sections | 2 bytes  | 0x0001-0xFFFF | uint16 denoting the number of the code sections                                    |
+| code_size         | 2 bytes  | 0x0001-0xFFFF | uint16 denoting the length of the code section content                             |
+| kind_data         | 1 byte   | 0x03          | kind marker for data size section                                                  |
+| data_size         | 2 bytes  | 0x0001-0xFFFF | uint16 integer denoting the length of the data section content                     |
```

Comment:
Because this proposes the data section to be mandatory we should allow a data section of size 0.